ssh/tailssh/incubator.go

```diff
@@ -699,9 +701,40 @@ func fileExists(path string) bool {
 	return err == nil
 }
 
+// loginArgs returns the arguments to use to exec the login binary.
+// It returns nil if the login binary should not be used.
+// The login binary is only used:
+//   - on darwin, if the client is requesting a shell or a command.
+//   - on linux and BSD, if the client is requesting a shell with a TTY.
 func (ia *incubatorArgs) loginArgs() []string {
+	if ia.isSFTP {
+		return nil
+	}
 	switch runtime.GOOS {
+	case "darwin":
+		args := []string{
+			ia.loginCmdPath,
+			"-f", // already authenticated
+			"-p", // discard previous environment, may change to -pq
```

Comment:
The comment seems to indicate the intent is to discard the previous environment, but `-p` tells `login` to preserve it instead.

Reply:
sorry, the comment is wrong. I'll send a PR to update it. We want to preserve the environment that we passed in like `$SHELL` (everything from `envForUser`)